### PR TITLE
Outdoor Directions using Google Directions API

### PR DESCRIPTION
**Pull request by makchamp**
-Click on a building then in the bottom-sheet click on the directions button
-It will generate and draw a path from your current location to the campus building
-The transport mode is set to walking (it is possible to add many more parameters to a directions request)
-When clicking on directions the camera moves to the current location and the bottom sheet is collapsed
-The HTTP request library used is Volley
-The response output format used is JSON
-Refactoring of MainActivity.kt

### CODE REVIEW

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt [hunk 1]
```diff
@@ -31,28 +35,42 @@ import com.google.android.gms.maps.model.Polygon
...
 import kotlinx.android.synthetic.main.bottom_sheet_layout.bottom_sheet
+import org.json.JSONObject
 import java.io.IOException
 import java.util.Locale
 
-
-//OnMapReadyCallback : interface ; extends AppCompatActivity() ;  GoogleMap.OnMarkerClickListener interface, which defines the onMarkerClick(), called when a marker is clicked or tapped:
-class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarkerClickListener,
-    GoogleMap.OnPolygonClickListener {
+class MapsActivity : AppCompatActivity(), OnMapReadyCallback,  GoogleMap.OnMarkerClickListener, GoogleMap.OnPolygonClickListener {
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: don't make your lines too long, it's a pain to side scroll to view code

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt [hunk 2]
```diff
@@ -31,28 +35,42 @@ import com.google.android.gms.maps.model.Polygon
...
+
+    companion object {
+        private const val LOCATION_PERMISSION_REQUEST_CODE = 1
+
+        private const val REQUEST_CHECK_SETTINGS = 2
+
+        private const val AUTOCOMPLETE_REQUEST_CODE = 3
+    }
+
+    private lateinit var bottomSheetBehavior : BottomSheetBehavior<View>
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: this whole diff has a space in between each line. Don't use spacing unless it provides logical separation between code functionalities

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt [hunk 3]
```diff
@@ -65,14 +83,13 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarker
...
 
-
         //update lastLocation with the new location and update the map with the new location coordinates
         locationCallback = object : LocationCallback() {
             override fun onLocationResult(p0: LocationResult) {
                 super.onLocationResult(p0)
 
                 lastLocation = p0.lastLocation
-                placeMarkerOnMap(LatLng(lastLocation.latitude, lastLocation.longitude))
+              //  placeMarkerOnMap(LatLng(lastLocation.latitude, lastLocation.longitude))
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: why is this commented out?

Comment (NicholasNagy) [reply app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: If we don't need it, remove it

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt [hunk 4]
```diff
@@ -365,6 +247,110 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarker
...
+                centerLong += p.points[i].longitude
+            }
+            centerLat /= p.points.size
+            centerLong /=p.points.size
+
+            val buildingLocation : Location = lastLocation
+            buildingLocation.latitude = centerLat
+            buildingLocation.longitude = centerLong
+
+            //TODO: This full clear and redraw should probably be removed when the directions system is implemented. It was added to show only one route at a time
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: make this multi-line please

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt [hunk 5]
```diff
@@ -365,6 +247,110 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarker
...
+                    addressText += if (i == 0) address.getAddressLine(i) else "\n" + address.getAddressLine(i)
+                }
+            }
+        } catch (e: IOException) {
+            Log.e("MapsActivity", e.localizedMessage!!)
+        }
+
+        return addressText
+    }
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: no need for double spacing

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt (hunk 1 above)
Comment (makchamp) [reply app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: Good point

File: app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt (hunk 3 above)
Comment (makchamp) [reply app/src/main/java/com/droidhats/campuscompass/MapsActivity.kt]: Yea we can get rid of it

**Pull request merged**